### PR TITLE
feat: full URL support for buildUrl and buildSrcSet

### PR DESCRIPTION
**Pull request by luqven**
This PR adds full URL support to the client via static methods

-  `_buildURL` 
- `_buildSrcSet`

Unit tests for these methods are included in the PR.

The PR also introduces a helper function, `extractUrl` that uses the `ufo` module to parse and isolate different URL components of a full URL. Unit tests for the helper function are also included in the PR.

The PR also introduces `ufo` [package](https://github.com/unjs/ufo) as a dependency.

## Examples

```js
const client = ImgixClient
const params = { w: 100 }
const opts = { useHttps: true }
const src = "sdk-test.imgix.net/amsterdam.jpg?h=100"

const url = client._buildURL(src, params, opts)
const srcSet = client._buildSrcSet(src, params, opts)

console.log(url) // => "https://sdk-test.imgix.net/amsterdam.jpg?h=100&w=100"
console.log(srcSet) /** =>
* https://sdk-test.imgix.net/amsterdam.jpg?w=800&q=100&dpr=1&s=f556bd48d069e6435d8f59940fbd2830 1x,
* https://sdk-test.imgix.net/amsterdam.jpg?w=100&h=100&dpr=2 2x,
* https://sdk-test.imgix.net/amsterdam.jpg?w=100&h=100&&dpr=3 3x,
* https://sdk-test.imgix.net/amsterdam.jpg?w=100&h=100&dpr=4 4x,
* https://sdk-test.imgix.net/amsterdam.jpg?w=100&h=100&dpr=5 5x"
*/
```

## Future Work
- There are edge cases that might be missing from the unit tests. Please comment on the PR if any come to mind.
- These methods could be exported as named exports to avoid having to call `client.{method}`. This can be added in a subsequent PR if needed.

## Comments

This PR is not considered a breaking change since none of the existing public methods are changed.

This PR includes work and feedback from:
- #287 
- #288 
- #290 

> Please note that support for full URLs is considered experimental and will not be officially supported.

### CODE REVIEW

**Review by sherwinski — COMMENTED** [review]
Well done! This really came together 👏 👏 

Just one discussion point before we merge this in.

File: src/index.js [hunk 1]
```diff
@@ -120,6 +168,42 @@ export default class ImgixClient {
...
+   * _buildSrcSet static method allows full URLs to be used when generating
+   * imgix formatted `srcset` string values.
+   *
+   * - If the source URL has included parameters, they are merged with
+   * the `params` passed in as an argument.
+   * - URL must match `{host}/{pathname}?{query}` otherwise an error is thrown.
+   *
+   * @param {String} url - full source URL path string, required
+   * @param {Object} params - imgix params object, optional
+   * @param {Object} options - imgix client options, optional
```
[the remark this answers is not in the document]
Comment (luqven) [reply src/index.js:181]: That ms. 

I wouldn't be against adding parameters to this method to separate client options from sourceset modifiers.

So that instead of

```js 
_buildSrcSet(url, params = {}, options = {})
```

The arguments could be something more like

```js
_buildSrcSet(url, params = {}, srcsetModifiers = {}, clientOptions = {})
```

Obv we'd update the js-doc to match, something like

```js
/**
   * @param {String} url - full source URL path string, required
   * @param {Object} params - imgix params object, optional
   * @param {Object} srcsetModifiers - srcset modifiers, optional
   * @param {Object} clientOptions - imgix client options, optional
*/
```

Would you be in favor of introducing those params @sherwinski ?

[the remark this answers is not in the document]
Comment (sherwinski) [reply src/index.js:181]: Yeah that would be most clear imo 👍

**Review by sherwinski — APPROVED** [review]
LGTM 👍

**Comment by luqven** [comment]
Forced pushed to try out semantic release on `@next`.

**Comment by luqven** [comment]
Ran into an issue with colliding deps. Had to rebase and force push again.
